admin/acceptance.py

```diff
@@ -1212,25 +1227,27 @@ def main(reactor, args, base_path, top_level):
         result = 1
         raise
     finally:
-        # Unless the tests failed, and the user asked to keep the nodes, we
-        # delete them.
-        if not options['keep']:
-            runner.stop_cluster(reactor)
+        reached_finally = True
+        # We delete the nodes if the user hasn't asked to keep them
+        # or if we failed to provision the cluster.
+        if not setup_succeeded:
+            print "cluster provisioning failed"
+        elif not options['keep']:
+            print "not keeping cluster"
         else:
             print "--keep specified, not destroying nodes."
-            if cluster is None:
-                print ("Didn't finish creating the cluster.")
-            else:
-                print ("To run acceptance tests against these nodes, "
-                       "set the following environment variables: ")
+            assert cluster is not None
+            print ("To run acceptance tests against these nodes, "
+                   "set the following environment variables: ")
 
-                environment_variables = get_trial_environment(cluster)
+            environment_variables = get_trial_environment(cluster)
 
-                for environment_variable in environment_variables:
-                    print "export {name}={value};".format(
-                        name=environment_variable,
-                        value=shell_quote(
-                            environment_variables[environment_variable]),
-                    )
+            for environment_variable in environment_variables:
+                print "export {name}={value};".format(
+                    name=environment_variable,
+                    value=shell_quote(
+                        environment_variables[environment_variable]),
+                )
+            stop_cluster = False
```

Comment:
If you want, instead of this `stop_cluster` flag, you could use `reactor.removeSystemEventTrigger`.  It takes as an argument the object returned by `addSystemEventTrigger`.